Update premium note (#130)
risingwavelabs/risingwave-docs: get-started/rw-premium-edition-intro.mdx

While the Premium Edition is a paid offering, it is designed to complement and enhance the open-source RisingWave Community Edition. The Community Edition will continue to be freely available under the Apache License (Version 2.0), and we remain committed to supporting the needs of our open-source community and ecosystem.

The RisingWave Community Edition and Premium Edition share the same underlying binary or container image. Users of the Community Edition need to purchase a license key to access the Premium Edition features.

> For RisingWave Cloud users, all Premium Edition features are available out of box without additional cost. 

Reviewer: ```suggestion
For RisingWave Cloud users, all Premium Edition features are available out of the box without additional cost. 
```

- Performance improvements for non-standard deployment environments.
- Tailored features specifically requested by our paying customers.

> The following are the Premium Edition features, which we have added "Premium Edition Feature" note in the documentation.

Reviewer: ```suggestion
The following are Premium Edition features, which include a "Premium Edition Feature" note in the documentation.
```
